src/fluree/db/conn/file.cljc

```diff
@@ -205,6 +211,12 @@
   (-ctx-read [conn context-key] (go (read-context conn context-key)))
   (-ctx-write [conn ledger context-data] (go (write-context conn ledger
                                                             context-data)))
+  (-index-file-write [conn ledger index-type index-data]
```

Comment:
This should probably wait for a later pull request, but I think we should start moving to much slimmer protocols with more primitive methods to do basic operations like accessing storage, and implementing higher level functionality in terms of those methods using plain functions that accept protocol implementations as arguments. I prefer to use protocols to wrap primitive, external interactions with stateful resources like an s3 bucket, file system, or some kind of external api. 

To make that concrete, in this case, I think we should have generic read and write methods on a storage record, and, if necessary, have different storage records attached to a connection for things like contexts and indexes, but only if those things need to be stored in different places. 

I'd like to keep protocols as close to the "metal" (or cloud, or api request, whatever) as possible, and implement our business logic functionality in term of those primitives. That will make the the protocols more general, less concerned with changing implementation details, and more composable.